README.md

```diff
@@ -364,6 +364,18 @@ with double quotes and sometimes it is necessary.
 
 - For `initscripts`,  `mac` is the currently configured MAC address of the device (`HWADDR`).
 
+### `cloned_mac`
+
+The `cloned_mac` address is optional and allow to specify the strategy to get the default
+mac or to set your own mac. The value of the `cloned_mac` address needs to be specified in
+hexadecimal notation like `mac` property. Besides explicitly specifying the value as a MAC
+address with hexadecimal notation, the following special values are also supported:
+
+- `permanent`: use the permanent MAC address of the device
+- `preserve`: don't change the MAC address of the device upon activation
+- `random`: generate a randomized value upon each connect
+- `stable`: generate a stable, hashed MAC address
```

Comment:
if you omit the new `cloned_mac` property in the playbook, then `ArgValidatorMac()` gives `None` and `connection["cloned_mac"]` will be `None`.

That translates to `ethernet.cloned-mac-address=` in nmcli (or NULL in C code of the daemon).

So beside what is documented here, there is yet another value in the playbook. Which is to omit `cloned_mac` and has a special meaning (that none of the other values covers): to honor the global connection default.


I think, there should be one more enum value:
```
- `permanent`: use the permanent MAC address of the device
- `preserve`: don't change the MAC address of the device upon activation
- `random`: generate a randomized value upon each connect
- `stable`: generate a stable, hashed MAC address
- `default`: honor the default behavior in NetworkManager, which can be configured
   in NetworkManager.conf or falls back to "preserve".

Omitting the `cloned_mac` key is the same as setting `default`.
```

The advantage is, that in the playbook you can now do:
```
    cloned_mac: "{{ cloned_mac }}"
```
where the variable can also mean to use the default. Well, I am not familiar with Jinja templates, so I suspect it's even possible with the current approach to clear the `cloned_mac` key from the playbook if `{{ cloned_mac }}` is unset. But that seems non-obvious (I wouldn't know the syntax for that). And in general it seems nice that all the possible values have an explicit string representation (`{{ cloned_mac }}`, including "default") and omitting the `cloned_mac:` key is the same as setting one of those values.


And below:
```
                ArgValidatorMac(
                    "cloned_mac",
                    default_value="default",
                    enum_values=["default", "preserve", "permanent", "random", "stable"],
                ),
```
(and handle `"default"` in the code).

Reply:
I do not necessarily agree, in my opinion, when we want the default conf we do not define it in jinja or according to [the documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#making-variables-optional) with something like: 
```
cloned_mac: "{{ cloned_mac | default(omit) }}"
```
@liangwen12year what do you think about it?

Reply:
> I do not necessarily agree, in my opinion, when we want the default conf 

Note that "the default conf" in this case is another mode that is not configurable otherwise.

For example, for "mtu" all valid values range from `0` to `0xFFFFFFFF` (whereas 0 means that NetworkManager does not configure the MTU). Indeed, omitting the "mtu" key in the playbook is the same as setting it to zero "0", so you could say "the mtu key in the playbook defaults to zero". It is the same as setting `mtu: 0`, and you could explicitly set the value that happens to be the default.

With the `cloned_mac` property as it is, omitting the key defaults to a certain behavior, which is not accessibly otherwise. Omitting the key is a special setting itself (and not a default for an explicitly configurable settings).

> cloned_mac: "{{ cloned_mac | default(omit) }}"

Ah right. I thought this would be possible to achieve. Seems non-obvious though.

Reply:
to be clear, we probably have properties in the playbook where omitting them is some genuine behavior. So if others also think this is fine, then I don't mind :)

Reply:
Honestly, I would prefer to add the `default` value into the enum values `enum_values=["default", "preserve", "permanent", "random", "stable"],` and the have the `default` value documented as "honor the default behavior in NetworkManager ... ". The benefit is that when omiting the `cloned_mac` setting, the `default` value can still be set in the controlled level but not at the user level with the hack  `cloned_mac: "{{ cloned_mac | default(omit) }}"`. Plus, all the possible value for the `cloned_mac` can be listed with explicit string represenation.

Reply:
Ok, I did a simple implementation that do nothing when `default` is set, is it what you expect ?

Reply:
I think so, when default is set, then the setting `cloned_mac` in the connection profile is unset,  then NM lookup the global connection defaults (in NetworkManager.conf). If it's not specified there either, then the current version will default to "preserve".